src/D2L.CodeStyle.Analyzers/ApiUsage/ServiceLocator/OldAndBrokenServiceLocatorAnalyzer.cs

```diff
@@ -30,105 +42,104 @@ bool excludeKnownProblemDlls
+				context => {
+					IInvocationOperation invocation = (IInvocationOperation)context.Operation;
+					AnalyzeMemberUsage( context, invocation.TargetMethod, typeRules );
+				},
+				OperationKind.Invocation
+			);
+
+			context.RegisterOperationAction(
+				context => {
+					IMethodReferenceOperation reference = (IMethodReferenceOperation)context.Operation;
+					AnalyzeMemberUsage( context, reference.Method, typeRules );
+				},
+				OperationKind.MethodReference
+			);
+
+			context.RegisterOperationAction(
+				context => {
+					IPropertyReferenceOperation reference = (IPropertyReferenceOperation)context.Operation;
+					AnalyzeMemberUsage( context, reference.Property, typeRules );
+				},
+				OperationKind.PropertyReference
 			);
 
 			context.RegisterSymbolAction(
-				ctx => PreventUnnecessaryAllowedListing(
-					ctx,
-					disallowedTypes,
-					allowedTypeList
-				),
+				ctx => PreventUnnecessaryAllowedListing( ctx, typeRules ),
 				SymbolKind.NamedType
 			);
 		}
 
-		//Prevent static usage of OldAndBrokenServiceLocator
-		//For example, OldAndBrokenServiceLocator.Instance.Get<IFoo>()
-		private void PreventOldAndBrokenUsage(
-			SyntaxNodeAnalysisContext context,
-			ImmutableArray<INamedTypeSymbol> disallowedTypes,
-			AllowedTypeList allowedTypeList
+		private void AnalyzeMemberUsage(
+			OperationAnalysisContext context,
+			ISymbol member,
+			TypeRuleSets typeRules
 		) {
-			if( !( context.Node is IdentifierNameSyntax syntax ) ) {
-				return;
-			}
 
-			if( !IdentifierIsOfDisallowedType( context.SemanticModel, disallowedTypes, syntax, context.CancellationToken ) ) {
+			if( !typeRules.Disallowed.Contains( member.ContainingType ) ) {
 				return;
 			}
 
-			var parentClasses = context.Node.Ancestors().OfType<TypeDeclarationSyntax>();
-			var parentSymbols = parentClasses.Select( c => context.SemanticModel.GetDeclaredSymbol( c, context.CancellationToken ) ).ToImmutableArray();
+			ISymbol caller = context.ContainingSymbol;
 
-			if( parentSymbols.Any( s => Attributes.DIFramework.IsDefined( s ) ) ) {
-				//Classes in the DI Framework are allowed to use locators and activators
+			ImmutableArray<INamedTypeSymbol> callerContainingTypes = caller
+				.GetAllContainingTypes( sortDeepestFirst: true );
+
+			// Allow the DI framework to call the disallowed types
+			if( callerContainingTypes.Any( Attributes.DIFramework.IsDefined ) ) {
 				return;
 			}
 
-			if( _excludeKnownProblems && parentSymbols.Any( allowedTypeList.Contains ) ) {
-				return;
+			if( m_excludeKnownProblems ) {
+
+				// Allow the types listed in OldAndBrokenServiceLocatorAllowedList.txt
+				if( callerContainingTypes.Any( typeRules.Allowed.Contains ) ) {
+					return;
+				}
 			}
 
-			context.ReportDiagnostic(
-				Diagnostic.Create( Diagnostics.OldAndBrokenLocatorIsObsolete, syntax.GetLocation() )
+			Diagnostic diagnostic = Diagnostic.Create(
+				Diagnostics.OldAndBrokenLocatorIsObsolete,
+				context.Operation.Syntax.GetLocation()

```

Comment:
Being lazy with the location.  I imagine people have stopped trying to use these old methods by now?  A little extra red might scare them more.